### PR TITLE
Fsamin/fix 2018 07 04 13 46 07

### PR DESCRIPTION
**Pull request by fsamin**
please dont't squash to keep the different commits

### CODE REVIEW

File: engine/api/worker/worker.go [hunk 1]
```diff
@@ -45,12 +45,32 @@ func DeleteWorker(db *gorp.DbMap, id string) error {
...
+	return nil
+}
+
+// DisableWorker disable a worker
+func DisableWorker(db *gorp.DbMap, id string) error {
+	tx, errb := db.Begin()
+	if errb != nil {
+		return fmt.Errorf("DisableWorker> Cannot start tx: %s", errb)
+	}
+	defer tx.Rollback()
```
Comment (golangci) [inline engine/api/worker/worker.go]: Error return value of `tx.Rollback` is not checked

File: sdk/cdsclient/client_queue.go [hunk 2]
```diff
@@ -20,10 +20,29 @@ import (
 	"github.com/ovh/cds/sdk"
 )
 
+func shkringQueue(queue *sdk.WorkflowQueue, l int) time.Time {
```
Comment (bnjjj) [inline sdk/cdsclient/client_queue.go]: shring
+ comment

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#4896.0 ✘  
* Build minimal
  * cds cli legacy ✔ 
  * cdsctl linux ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔ 
* Build other os/arch
  * cds cli legacy darwin/windows amd64 -  
  * cdsctl darwin -  
  * cdsctl freebsd amd64 -  
  * Doc Build -  
  * Engine Build darwin amd64 -  
  * Engine Build freebsd amd64 -  
  * Engine Build linux arm -  
  * Engine Build windows amd64 -  
  * Plugins darwin amd64 -  
  * Plugins freebsd -  
  * Plugins linux amd64 -  
  * Plugins linux arm -  
  * Plugins windows amd64 -  
  * Worker darwin / windows / freebsd -  
  * Worker linux arm -  
* Package
  * Debian Engine Package ✔ 
  * Docker Engine Package ✘

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#4896.1 ✘  
* Build minimal
  * cds cli legacy ✔ 
  * cdsctl linux ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔ 
* Build other os/arch
  * cds cli legacy darwin/windows amd64 -  
  * cdsctl darwin -  
  * cdsctl freebsd amd64 -  
  * Doc Build -  
  * Engine Build darwin amd64 -  
  * Engine Build freebsd amd64 -  
  * Engine Build linux arm -  
  * Engine Build windows amd64 -  
  * Plugins darwin amd64 -  
  * Plugins freebsd -  
  * Plugins linux amd64 -  
  * Plugins linux arm -  
  * Plugins windows amd64 -  
  * Worker darwin / windows / freebsd -  
  * Worker linux arm -  
* Package
  * Debian Engine Package ✔ 
  * Docker Engine Package ✘

**Pull request merged**